src/ligero/mod.rs

```diff
@@ -344,53 +343,72 @@ where
         labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F, P>>,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
         point: &'a P::Point,
-        challenge_generator: &mut crate::challenge::ChallengeGenerator<F, S>,
-        rands: impl IntoIterator<Item = &'a Self::Randomness>,
-        rng: Option<&mut dyn RngCore>,
+        _challenge_generator: &mut crate::challenge::ChallengeGenerator<F, S>,
+        _rands: impl IntoIterator<Item = &'a Self::Randomness>,
+        _rng: Option<&mut dyn RngCore>,
     ) -> Result<Self::Proof, Self::Error>
     where
         P: 'a,
         Self::Randomness: 'a,
         Self::Commitment: 'a,
     {
-        let labeled_polynomial = labeled_polynomials.into_iter().next().unwrap();
-        let polynomial = labeled_polynomial.polynomial();
+        let mut proof_array = LigeroPCProofArray::new();
+        let labeled_commitments: Vec<&'a LabeledCommitment<Self::Commitment>> =
+            commitments.into_iter().collect();
+        let labeled_polynomials: Vec<&'a LabeledPolynomial<F, P>> =
+            labeled_polynomials.into_iter().collect();
```

Comment:
this looks a little strange, couldn't we loop over the `labeled_polynomials` directly, instead of first forcing it into a vector?

Reply:
Turning it into a vector is necessary to check its length, which is done mostly for the initial consistency checks. The length check definitely can't be done on an `impl IntoIterator`, and if we only only convert it to an iterator (instead of collecting it into a vector) then checking its length consumes it and we need to convert it again when we want to loop over it - at which point we might as well convert it into a vector from the beginning. Does this sound convincing or am I missing some obvious better alternative?

Reply:
I see what you mean. I wonder if using `zip` could be used as an alternative, and then at the end ensuring that `next()` panics on both iters - probably not worth it though for now. Let's merge it